app/vmselect/vmui.go

```diff
@@ -0,0 +1,148 @@
+type panelSettings struct {
+	Title       string   `json:"title,omitempty"`
+	Description string   `json:"description,omitempty"`
+	Unit        string   `json:"unit,omitempty"`
+	Expr        []string `json:"expr"`
+	Alias       []string `json:"alias,omitempty"`
+	ShowLegend  bool     `json:"showLegend,omitempty"`
+	Width       int      `json:"width,omitempty"`
+}
+
+// dashboardRow represents panels on dashboard
+// fields of the dashboardRow you can find by following next link
+// https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/app/vmui/packages/vmui/public/dashboards
+type dashboardRow struct {
+	Title  string          `json:"title,omitempty"`
+	Panels []panelSettings `json:"panels"`
+}
+
+// dashboardsData represents all dashboards settings
+type dashboardsData struct {
+	DashboardsSettings []dashboardSetting `json:"dashboardsSettings"`
+
+	mx                sync.Mutex
+	expireAtTimestamp time.Time
+}
+
+func handleVMUICustomDashboards(w http.ResponseWriter) {
+	path := *vmuiCustomDashboardsPath
+	if path == "" {
+		writeSuccessResponse(w, predefinedDashboards)
+		return
+	}
+
+	if !fs.IsPathExist(path) {
+		writeErrorResponse(w, fmt.Errorf("cannot find folder pointed by -vmui.customDashboardsPath=%q", path))
+		return
+	}
+
+	requestTime := time.Now()
+	if predefinedDashboards.expireAtTimestamp.Before(requestTime) {
+		settings, err := collectDashboardsSettings(path)
+		if err != nil {
+			writeErrorResponse(w, fmt.Errorf("cannot collect dashboards settings by -vmui.customDashboardsPath=%q", path))
+			return
+		}
+
+		predefinedDashboards.expireAtTimestamp = time.Now().Add(updateDashboardsTimeout)
+
+		predefinedDashboards.mx.Lock()
+		predefinedDashboards.DashboardsSettings = nil
+		predefinedDashboards.DashboardsSettings = append(predefinedDashboards.DashboardsSettings, settings...)
+		predefinedDashboards.mx.Unlock()
+	}
+
+	writeSuccessResponse(w, predefinedDashboards)
+}
+
+func writeErrorResponse(w http.ResponseWriter, err error) {
+	w.WriteHeader(http.StatusBadRequest)
+	w.Header().Set("Content-Type", "application/json")
+	fmt.Fprintf(w, `{"status":"error","error":"%q"}`, err.Error())
+}
+
+func writeSuccessResponse(w http.ResponseWriter, data *dashboardsData) {
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(data); err != nil {
+		logger.Errorf("error encode dashboards settings: %s", err)
+	}
+}
+
+func collectDashboardsSettings(path string) ([]dashboardSetting, error) {
+
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read folder pointed by -vmui.customDashboardsPath=%q", path)
+	}
+
+	var settings []dashboardSetting
+	for _, file := range files {
+		info, err := file.Info()
+		if err != nil {
+			logger.Errorf("cannot get file info: %s", err)
+			continue
+		}
+		if fs.IsDirOrSymlink(info) {
+			logger.Infof("skip directory or symlinks in the -vmui.customDashboardsPath=%q", path)
```

Comment:
This message doesn't contain information on which directory or symlink is skipped :(

Reply:
added file info